android/java/org/chromium/chrome/browser/crypto_wallet/fragments/NftGridFragment.java

```diff
@@ -0,0 +1,335 @@
+        return view;
+    }
+
+    // TODO: refactor this entire method once multi-chain support will be merged.
+    private void setUpObservers() {
+        if (mWalletModel == null) return;
+        mWalletModel.getCryptoModel().getNetworkModel().mCryptoNetworks.observe(
+                getViewLifecycleOwner(),
+                allNetworkInfos -> { mAllNetworkInfos = allNetworkInfos; });
+        mWalletModel.getCryptoModel().getNetworkModel().mDefaultNetwork.observe(
+                getViewLifecycleOwner(), networkInfo -> {
+                    if (networkInfo == null) return;
+                    if (mNetworkInfo != null && !mNetworkInfo.chainId.equals(networkInfo.chainId)) {
+                        // Clean up list to avoid user clicking on an asset of the previously
+                        // selected network after the network has been changed.
+                        clearAssets();
+                    }
+                    mNetworkInfo = networkInfo;
+                    updateNftGrid();
+                });
+        mPortfolioModel.mNftModels.observe(getViewLifecycleOwner(), nftDataModels -> {
+            if (mPortfolioModel.mPortfolioHelper == null) return;
+            mNftDataModels = nftDataModels;
+            setUpNftList(nftDataModels, mPortfolioModel.mPortfolioHelper.getPerTokenCryptoSum(),
+                    mPortfolioModel.mPortfolioHelper.getPerTokenFiatSum());
+        });
+
+        mPortfolioModel.mIsDiscoveringUserAssets.observe(
+                getViewLifecycleOwner(), isDiscoveringUserAssets -> {
+                    if (isDiscoveringUserAssets) {
+                        mPbAssetDiscovery.setVisibility(View.VISIBLE);
+                    } else {
+                        mPbAssetDiscovery.setVisibility(View.GONE);
+                        updateNftGrid();
+                    }
+                });
+
+        mWalletModel.getCryptoModel().getNetworkModel().mNeedToCreateAccountForNetwork.observe(
+                getViewLifecycleOwner(), networkInfo -> {
+                    if (networkInfo == null) return;
+                    MaterialAlertDialogBuilder builder =
+                            new MaterialAlertDialogBuilder(
+                                    requireContext(), R.style.BraveWalletAlertDialogTheme)
+                                    .setMessage(getString(
+                                            R.string.brave_wallet_create_account_description,
+                                            networkInfo.symbolName))
+                                    .setPositiveButton(R.string.wallet_action_yes,
+                                            (dialog, which) -> {
+                                                mWalletModel.getCryptoModel()
+                                                        .getNetworkModel()
+                                                        .setNetwork(networkInfo, success -> {
+                                                            if (success) {
```

Comment:
Perhaps makes sense to split it to a few functions to avoid such a construction

Reply:
Yes, I'll split this logic into multiple functions and try to make it more readable.